### PR TITLE
topics: standardize `strings` topic

### PR DESCRIPTION
**Pull request by azul**
There were quite a few topics that were string related
and not listed in the [topics list](https://github.com/exercism/problem-specifications/blob/master/TOPICS.txt).

* Replace all of them with `strings` which is in the list.
* Add `parsing` if the topic was `string_parsing`.

### CODE REVIEW

**Comment by azul** [comment]
working towards #529

File: config.json [hunk 1]
```diff
@@ -308,7 +307,7 @@
       "topics": [
         "match",
         "result",
-        "str_vs_string",
```
Comment (coriolinus) [inline config.json]: I think we should not remove the `str_vs_string` topic from any exercises. While that's not in the canonical topics list, it's an important distinction that is unique to Rust; it's useful to retain even if we also add the `strings` topic.

**Comment by azul** [comment]
I made the changes @coriolinus requested and pushed again.

File: config.json [hunk 2]
```diff
@@ -636,7 +640,7 @@
         "chars",
         "iterators",
         "primitive_types",
-        "str_vs_string"
```
Comment (coriolinus) [inline config.json]: You're still removing `str_vs_string` here, which I think should remain.

File: config.json [hunk 3]
```diff
@@ -648,7 +652,7 @@
       "topics": [
         "ascii",
         "chars",
-        "str_vs_string"
```
Comment (coriolinus) [inline config.json]: Here also.

**Comment by azul** [comment]
@coriolinus thanks for pointing them out. Should be fixed now.

**Pull request merged**